[이호현] Add: Monsters.js, CardList.js, Card.js 작업

LeeHoHyun-hemtory opened the pull request:
wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- Monsters.js, CardList.js 내용 수정했습니다
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- Monsters.js: arrow function return, { } 생략
- Monsters.js: 변수명 이상한거 변경
- CardList.js: arrow function return, { } 생략

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [X] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [X] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [X] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

jamessoun93 commented on src/2-monsters/index.js:
```diff
@@ -21,26 +21,38 @@ import "./index.css";
+	componentDidMount() {
+		fetch('https://jsonplaceholder.typicode.com/users')
+			.then(response => response.json())
+			.then(json => this.setState({ monsters: this.state.monsters.concat(json) }));
```
여기서 인자로 받는 데이터는 JSON 데이터가 아니기 때문에 헷갈릴 수 있는 이름은 피해주시는게 좋습니다. 그리고,
``` js
this.setState({ monsters: this.state.monsters.concat(json) });
```
는 기존 monsters 배열과 들어온 json 배열을 합친 새로운 배열을 setState 해주시는 부분인데요. 문제는 없습니다만, 굳이 빈 배열을 합쳐주는 것 보다는 들어온 배열로 아예 바꿔치기 하시는 편이 좋아보입니다.
``` js
this.setState({ monsters: json });
```
이렇게요!

jamessoun93 commented on src/2-monsters/index.js:
```diff
@@ -21,26 +21,38 @@ import "./index.css";
+	render() {
+		// 필터링 로직
+		const newMonsters = this.state.monsters.filter(monster =>
+			monster.name.toLowerCase().includes(this.state.userInput)
```
더욱 더 정확한 비교를 위해 현재 비교대상인 monster.name 과 userInput 을 둘다 소문자 형태로 변환해주는 것이 좋습니다.

LeeHoHyun-hemtory committed:
Add: Monsters.js, CardList.js, Card.js 작업
- Monsters.js: fetch기능 추가
- Monsters.js: 이벤트 핸들러 추가
- Monsters.js: SearchBox filter 기능 추가
- CardList.js: Card map method 기능 추가
- Card.js: props 값 사용

LeeHoHyun-hemtory committed:
Modify: SearchBox.js 수정
- input 지우면 CardList 원복

jamessoun93 commented on src/Assignments/Monsters/students/11th/LeeHoHyun-hemtory/Components/Card/Card.js:
```diff
@@ -22,11 +22,11 @@ class Card extends Component {
-          src="https://live.staticflickr.com/4561/38054606355_26429c884f_b.jpg"
+          src={`https://robohash.org/${this.props.id}?set=set2&size=180x180`}
           alt=""
         />
-        <h2>Name</h2>
-        <p>Email</p>
+        <h2>{this.props.name}</h2>
+        <p>{this.props.email}</p>
```
👍 👍 👍

jamessoun93 commented on src/Assignments/Monsters/students/11th/LeeHoHyun-hemtory/Components/CardList/CardList.js:
```diff
@@ -11,7 +11,11 @@ import "./CardList.scss";
-    return <div className="card-list"></div>;
+    return <div className="card-list">
+      {this.props.monsters.map((mon) => {
+        return <Card key={mon.id} id={mon.id} name={mon.name} email={mon.email}/>
```
중괄호와 return 문은 생략할 수 있겠죠? 😇

LeeHoHyun-hemtory replied:
Card component에 빨간줄 그어지길래 return이 빠져서 그런가했는데
( )로 묶어주고 { } 생략하니까 잘 적용되네요
수정하겠습니다

jamessoun93 commented on src/Assignments/Monsters/students/11th/LeeHoHyun-hemtory/Monsters.js:
```diff
@@ -27,15 +27,28 @@ class Monsters extends Component {
   };
 
   // 데이터 로딩
+  componentDidMount(){
+    fetch('https://jsonplaceholder.typicode.com/users')
+    .then(reponse => reponse.json())
+    .then(json => this.setState({monsters: json}));
+  }
```
👍👍👍

jamessoun93 commented on src/Assignments/Monsters/students/11th/LeeHoHyun-hemtory/Monsters.js:
```diff
@@ -27,15 +27,28 @@ class Monsters extends Component {
   render() {
+    let temp = this.state.monsters.filter(mon => {
+      return mon.name.toLowerCase().includes(this.state.userInput.toLowerCase());
+    });
```
필터링 로직 좋습니다! 하지만 이 부분도 단순 리턴문이기 때문에 중괄호와 return 키워드를 생략할 수 있겠죠?

jamessoun93 replied:
그리고 temp 보다는 조금 더 명확한 변수명을 사용해주셨으면 좋겠습니다!

LeeHoHyun-hemtory replied:
수정하겠습니다

jamessoun93 reviewed (changes requested):
리뷰내용 확인해주세요!

LeeHoHyun-hemtory committed:
Modify: Monsters.js, CardList.js 내용 수정
- Monsters.js: arrow function return, {} 생략
- Monsters.js: 변수명 수정
- CardList.js: arrow function return, {} 생략

jamessoun93 commented:
수고하셨습니다!

LGTM! 🙇🏻‍♂️
